libraries/AP_Math/definitions.h

```diff
@@ -0,0 +1,74 @@
+#pragma once
+
+#include <math.h>
+#include <AP_HAL/AP_HAL.h>
+
+
+// Double precision math must be activated
+#if defined(DBL_MATH) && CONFIG_HAL_BOARD == HAL_BOARD_LINUX
+  #ifndef M_PI
+    #define M_PI      (3.14159265358979323846)
+  #endif
+
+  #ifndef M_PI_2
+    #define M_PI_2    (M_PI / 2)
+  #endif
+#else  		// Standard single precision math
+  #ifdef M_PI
+    #undef M_PI
+    #define M_PI      (3.141592653589793f)
```

Comment:
I moved the define to be outside of the ifdef. Otherwise it doesn't build on toolchains without M_PI. We always want to override its definition anyway.